controllers/events.js

```diff
@@ -0,0 +1,207 @@
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The events data in JSON format.
+ * @throws {Error} If an error occurs while retrieving the events data.
+ */
+const getAllEvents = async (req, res) => {
+  /**
+   * @type {boolean} - enabled: determines whether the rooms should be enabled or disabled.
+   * @type {number} - limit: The maximum number of rooms to retrieve.
+   * @type {string} - offset: The starting point for retrieving rooms.
+   */
+  const { enabled, limit, offset } = req.query;
+  try {
+    const start = offset || "";
+    const limitOfRooms = limit || GET_ALL_EVENTS_LIMIT_MIN;
+    const isEnabled = enabled || false;
+    const eventsData = await apiService.get(`/rooms?limit=${limitOfRooms}&enabled=${isEnabled}&start=${start}`);
+    if (eventsData.data) {
+      const propertiesToRemove = ["customer_id", "app_id", "recording_info", "template_id", "template", "customer"];
+      const events = removeUnwantedProperties(propertiesToRemove, eventsData.data);
+
+      const filteredEventsData = {
+        limit: eventsData.limit,
+        last: eventsData.last,
+        data: events.map(({ id, ...event }) => ({
+          id,
+          room_id: id,
+          ...event,
+        })),
+      };
+      return res.status(200).json(filteredEventsData);
+    }
+    return res.status(200).json(eventsData);
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't get events. Please try again later",
+    });
+  }
+};
+
+/**
+ * Generates a token for the specified event and user information.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} An object containing the generated token and a success message in JSON format.
+ * @throws {Error} If an error occurs while generating the token.
+ */
+const joinEvent = async (req, res) => {
+  const { roomId, userId, role } = req.body;
+  const payload = { room_id: roomId, user_id: userId, role };
+  try {
+    const token = tokenService.getAuthToken(payload);
+    res.status(201).json({
+      token: token,
+      message: "Token generated successfully!",
+      success: true,
+    });
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+};
+
+/**
+ * Retrieves event details by ID and returns JSON response
+ *
+ * @async
+ * @function
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ * @returns {Object} - JSON response containing event details or error message
+ * @throws {Object} - The JSON response with an error message if an error occurred if event retrieval fails.
+ */
+const getEventById = async (req, res) => {
+  const roomId = req.params.id;
+  const isActiveRoom = req.body.isActiveRoom;
```

Comment:
this would break on production in case body is empty 
either add optional chaining or a fallback object whatever needed 
same goes for params.id

ideally when we access something like this it works fine until we receive that data but in case something goes down all dependent services which don't have a guard rail handled start breaking

Reply:
We are using `eventsValidator.getEventById` middleware to validate both the `params` and `body`. So, if the body or the params are undefined the request will return 400 from there itself.